client/js/jsonedit.js

```
@@ -830,35 +897,89 @@ function displayComputeOps() {
-        call: async function() {
-          jePasteText(css + '; ', true);
+        id: 'css_' + cssSection,
+        name: cssSection,
+        context: `^${type} ↦ css`,
+        show:  function() {
+          if(typeof jeStateNow.css != "object" || JSON.stringify(jeStateNow.css) == '{}')
+            return false;
+          for(const property in jeStateNow.css)
+            if(typeof jeStateNow.css[property] != "object")
+              return false;
+          return jeStateNow.css[cssSection] == undefined ? true : false;
         },
-        show: function() {
-          return !String(jeGetValue()[jeGetLastKey()]).match(css.split(':')[0]);
+        call: async function() {
+          jeStateNow.css[cssSection] = '###SELECT ME###';
+          jeSetAndSelect({});
         }
       });
+      for(const cssProperty of Object.keys(nested_presets[type][cssSection])) { // Add entries per-section
+        jeCommands.push({
+          id: 'css_' + cssSection + '_' + cssProperty,
+          name: cssProperty,
+          context: `^${type} ↦ css ↦ [^↦]*`,
+          show: function() {
+            const contents = jeStateNow.css[cssSection];
+            return typeof contents == "object" && jeContext.includes(cssSection) && !(cssProperty in contents);
+          },
+          call: async function() {
+            jeStateNow.css[cssSection][cssProperty] = '###SELECT ME###';
+            jeSetAndSelect(nested_presets[type][cssSection][cssProperty]);
+          }
+        });
+      }
     }
   }
+
+  // Add simple object button items
+  for(const cssProperty of Object.keys(string_presets)) { // Add entries in "default" (only) section
+    jeCommands.push({
+      id: 'css_string_' + cssProperty,
+      name: cssProperty,
+      context: `.* ↦ (css|[a-z]+CSS)`,
+      show: function() { // Need to make sure it is a simple object (contents are "key": "string" pairs)
+        const cssKind = jeContext.join(' ↦ ').match(this.context)[1];
```

This should throw an exception every time the context does _not_ match. It should be caught and everything should be fine. I don't think I mind.

---

But it should always match, since that's how we got here.